Improve pre-publish panel styling

sarahmonster opened the pull request:
## Before
This resolves #7878 and resolves #7873, by clarifying the microcopy used in the prompt here, and adjusting the styling to be more consistent with other sidebars in the UI.

<img width="973" alt="screenshot 2018-07-10 20 44 31" src="https://user-images.githubusercontent.com/376315/42539376-d9f05656-8492-11e8-83fb-6cd9ae115d1c.png">


## After
<img width="1130" alt="screenshot 2018-07-10 22 35 26" src="https://user-images.githubusercontent.com/376315/42539361-cab8b232-8492-11e8-8b82-f65fe1442ebc.png">


## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->


## Types of changes
Style & copy tweak.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->

sarahmonster committed:
Use clearer wording for pre-publish panel prompt; fixes #7873.

sarahmonster committed:
Style pre-publish panel more consistently with settings sidebar; fixes #7878.

tofumatt commented on editor/components/post-publish-panel/style.scss:
```diff
@@ -44,22 +45,25 @@
 
 .editor-post-publish-panel__link {
 	color: $blue-medium-700;
+	font-weight: 400;
```
Any chance we have variables for these, or are they hard-coded everywhere? 🤷‍♂️

sarahmonster replied:
Nope! Hard-coded everywhere. There aren't that many font-weights technically available, although an audit could be worth doing at some point during the high-shine polish stage.

tofumatt commented on editor/components/post-publish-panel/style.scss:
```diff
@@ -44,22 +45,25 @@
 
 		.components-panel__body-toggle {
-			font-weight: 400;
+			//font-weight: 400;
```
🔥

sarahmonster replied:
🙈

tofumatt commented on editor/components/post-publish-panel/prepublish.js:
```diff
@@ -26,7 +26,7 @@ function PostPublishPanelPrepublish( {
 		<div className="editor-post-publish-panel__prepublish">
 			<div><strong>{ hasPublishAction ? __( 'Are you ready to publish?' ) : __( 'Are you ready to submit for review?' ) }</strong></div>
-			<p>{ hasPublishAction ? __( 'Here, you can do a last-minute check up of your settings below, before you publish.' ) : __( 'When you\'re ready, submit your work for review, and an Editor will be able to approve it for you.' ) }</p>
+			<p>{ hasPublishAction ? __( 'Double-check your settings, then use the button to publish your post.' ) : __( 'When you\'re ready, submit your work for review, and an Editor will be able to approve it for you.' ) }</p>
```
"the button" is a bit vague. Maybe `the "Publish" button`? But that's awkward. Why not:

> Double-check your settings, then click "Publish".

Short and to-the-point.

sarahmonster replied:
"Click" is too desktop-centric, hence the vague wording—I'd like to stick with "use" for lack of a better term.

tofumatt reviewed (commented):
This looks/reads better. I have a few questions/thoughts about it though.

sarahmonster committed:
Nix commented-out CSS.

karmatosed commented:
Interesting ideas @sarahmonster. I do think having this section look so close to the sidebar for document/block maybe is a step too far. Could we scale back have it reflect the design over be exact? Maybe we have the top section white for example. I think this is better as whilst it is a sidebar it isn't the same type of element. The content is different.

tofumatt commented:
Making the top section white makes everything very low contrast and makes it hard to single out where the user should be looking. The most important info (eg: what to do next and why this modal appeared) is highlighted by the darker background in the new design.

Comparing the two screenshots my eyes immediately know where to look in the one with the dark grey background over the white, where my eyes just dart all over the place.

sarahmonster commented:
For reference, this is what it looks like with the header section in white:

<img width="1132" alt="screenshot 2018-07-12 09 29 04" src="https://user-images.githubusercontent.com/376315/42622409-388b9286-85b8-11e8-9cd9-f9cd5aa12409.png">

For readability and usability, I think the most important thing here is to retain the visual separation between panels. 

I think the trouble here is that this sidebar pattern looks and functions almost exactly the same way as that other (pretty ubiquitous) sidebar pattern, with a few notable deviations. This makes it feel as though the difference here is accidental, rather than deliberate. There's also an odd visual thing where you end up with a very pronounced crosshair of grid lines in the top right of the screen, which commands focus.

If we don't want it to feel like the same element, a better solution might be to consider using a different element here, rather than a well-established pattern with a slight deviation. A drop-down from the interim publish button could work, and it's part of my suggestions for improvements in #7602.

karmatosed commented:
I actually think the white sidebar works. The problem with a drop down is extending, what is someone puts a check in there, or 10?

tofumatt commented:
> The problem with a drop down is extending, what is someone puts a check in there, or 10? 

It would scroll, just like it would need to if 10 checkboxes were in there on mobile. Though putting ten checkboxes in there is probably a UX bug 😉

I think the main issue with a dropdown that extends out of the publish button actually is that it's still not very attention-grabbing unless we blur/darken the rest of the editor. Which brings me to...

> I actually think the white sidebar works.

While it looks visually "clean", I don't think it draws any attention to itself. I already find the pre-publish sidebar modal a bit strange as I expect the post to publish when I click publish–but if it isn't we should be clear about what to focus on.

If you want to keep the sidebar white, how about we grey out the editor?

![untitled](https://user-images.githubusercontent.com/90871/42658574-9ea2f2b6-861d-11e8-82df-24f07c7629ba.png)

karmatosed commented:
I still feel for the format we have and patterns it being white makes sense. However as in #7602 other explorations may totally change this area. If we want an update to it right now then white would be the right pattern.

tofumatt commented:
As mentioned: fine with it being white, but having more visual draw when it's the last step between writing and publishing is better than what we have now, where a user's eyes wander while they wonder why "Publish" didn't publish 😄

All white plus the greyed out editor (as you shouldn't be making editor changes at this point anyway) would be a good idea I think (and probably easy to implement and work well with stuff mentioned in #7879).

sarahmonster committed:
Switch to an all-white background for prepublish panel.

sarahmonster commented:
I just realised this PR went stale, but in reviewing #8235 realised it's super confusing having the panels without any delineating border. 

In the interests of pushing this through, I've switched the background to white. It'll likely change more as we iterate through the changes in #7602, but for the moment this is an improvement on what we currently have.

gziolo commented:
I think originally @jasmussen and @youknowriad  shaped how this panel looks and behaves in general. If we want to make it look closer to other sidebars, we should consider refactoring it in a way that avoids maintaining two different solutions for sidebars.

sarahmonster commented:
Just to clarify, since there's been a bit of back-and-forth: this isn't an attempt to make it look more "sidebar-like", we've just adjusted the text strings and added back the separators between panels within the sidebar itself. This isn't intended to be a final solution, but rather a smaller iterative change to make this panel easier to use whilst we work on larger improvements to the overall flow.

Longer term, it'll likely be moving to using a different pattern entirely that's even less sidebar-ish than its current incarnation. 😄 For extensive discussion and preliminary mockups, see #7602.

gziolo commented:
Cool, thanks for clarifications @sarahmonster 👍 

I just wanted to avoid having code duplication which ends up looking the same for the end user. Nothing to worry about in such case and sadly no space to remove some code 😃

sarahmonster commented:
I regret to inform you I am 100% planning on adding more code before we're all finished here. 😬

jasmussen commented:
On this one, designwise, I'll defer to Tammie, I know she has lots of ideas around this one.

tofumatt reviewed (approved):
Code here looks good and I think this looks way better.

karmatosed reviewed (approved):
Let's try and get this in as a first step :)

sarahmonster commented:
Let's do it. 

:party-sheep:!

mtias commented:
Thanks for the improvements here!
